homeassistant/components/cover/mqtt.py

```diff
@@ -44,25 +44,24 @@
     vol.Optional(CONF_STATE_CLOSED, default=STATE_CLOSED): cv.string,
     vol.Optional(CONF_OPTIMISTIC, default=DEFAULT_OPTIMISTIC): cv.boolean,
     vol.Optional(CONF_RETAIN, default=DEFAULT_RETAIN): cv.boolean,
-
 })
 
 
-def setup_platform(hass, config, add_devices_callback, discovery_info=None):
-    """Add MQTT Cover."""
-    add_devices_callback([MqttCover(
+def setup_platform(hass, config, add_devices, discovery_info=None):
+    """Setup the MQTT Cover."""
+    add_devices([MqttCover(
         hass,
-        config[CONF_NAME],
+        config.get(CONF_NAME),
         config.get(CONF_STATE_TOPIC),
-        config[CONF_COMMAND_TOPIC],
-        config[CONF_QOS],
-        config[CONF_RETAIN],
-        config[CONF_STATE_OPEN],
-        config[CONF_STATE_CLOSED],
-        config[CONF_PAYLOAD_OPEN],
-        config[CONF_PAYLOAD_CLOSE],
-        config[CONF_PAYLOAD_STOP],
-        config[CONF_OPTIMISTIC],
+        config.get(CONF_COMMAND_TOPIC),
```

Comment:
FYI, although not a big difference, there is a difference between using square brackets `[…]` versus `get(…)`: the square bracket method will blow up if the key doesn't exist while `get(…)` fails silently. Because of that, my preference is to use the square brackets unless we want the code to not blow up if the key is not there.